### PR TITLE
fixes #8913 - add back content view options for content view version promote

### PR DESCRIPTION
**Pull request by jlsherrill**
after apipie changes made content_view_id optional

### CODE REVIEW

File: lib/hammer_cli_katello/id_resolver.rb [hunk 1]
```diff
@@ -72,6 +73,16 @@ def content_view_version_id(options)
...
       end
+
+      begin
+        options[key_content_view_id] ||= content_view_id(
+          scoped_options("content_view", options)
+        )
+      rescue HammerCLIForeman::MissingSeachOptions # rubocop:disable HandleExceptions
+        # Intentionally suppressing the exception,
+        # content_view is not always required.
+      end
```
Comment (komidore64) [inline lib/hammer_cli_katello/id_resolver.rb]: this `L81-84` block feels a little fishy. is there a way you could accomplish this without using `begin...rescue`? just feels dirty.

Comment (jlsherrill) [inline lib/hammer_cli_katello/id_resolver.rb]: Without changing HammerCLIForeman's id_resolver, i don't believe so.  This is used in other places.

Comment (komidore64) [inline lib/hammer_cli_katello/id_resolver.rb]: ah, okay.

**Comment by komidore64** [comment]
looks sane to me! ack

**Pull request merged**